python/foxglove-sdk/src/websocket_server.rs

```
@@ -140,23 +140,84 @@ impl PyWebSocketServer {
     /// Sets a new session ID and notifies all clients, causing them to reset their state.
     /// If no session ID is provided, generates a new one based on the current timestamp.
     #[pyo3(signature = (session_id=None))]
-    pub fn clear_session(&self, session_id: Option<String>) -> PyResult<()> {
-        if let Some(server) = &self.0 {
-            server.clear_session(session_id);
+    pub fn clear_session(&self, session_id: Option<String>) {
+        let Some(server) = &self.0 else {
+            return;
+        };
+        server.clear_session(session_id);
+    }
+
+    /// Publishes the current server timestamp to all clients.
+    pub fn broadcast_time(&self, timestamp_nanos: u64) {
+        let Some(server) = &self.0 else {
+            return;
+        };
+        server.broadcast_time(timestamp_nanos);
+    }
+
+    /// Send a status message to all clients.
+    pub fn publish_status(&self, status: &PyStatus) {
+        let Some(server) = &self.0 else {
+            return;
+        };
+        server.publish_status(status.0.clone());
+    }
+
+    /// Remove status messages by id from all clients.
+    pub fn remove_status(&self, status_ids: Vec<String>) {
+        let Some(server) = &self.0 else {
+            return;
+        };
+        server.remove_status(status_ids);
+    }
```

A bigger question about the API. Is it better to silently fail after the server is closed, or to throw an exception?

Since the default is to only close the server at exit, this seems ok to me as-is.

---

Either way, we should probably document that behavior.

It would be user-friendlier to not raise an exception, so that users aren't forced to litter their code with `with contextlib.suppress(...):` statements, but my rusty side says it's better to explicitly raise errors instead of trying to guess what the user was trying to do. Since we're trying to gear our python SDK towards user-friendliness, I suppose I agree with Dan.